### PR TITLE
Implement markdown

### PR DESCRIPTION
**Pull request by BenPearo**
In this branch, I updated the markdown parser to a current (and working) standard. I created a new layout for pages, so new pages can be created purely in markdown with the same format as all the old posts. I also created a default-legacy layout and applied it to all the old posts that used default before, so now we can alter the default to fit whatever our markdowny needs are.

You can test this by creating a new file in _posts. Name it `2020-10-9-page-title.md`. This will be the visible date for the page, and the end of the URL for that page.
You can just copy-paste this in the file to take a look
```---
layout: post
title:  "Markdown Support!"
categories: [events, gcc]
---

# Markdown baby!!!!
## Check these headers
Juicy content
```

It should look like this :)
Screen Shot 2020-10-09 at 10 58 03 PM

### CODE REVIEW

File: _layouts/post.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                    <article class="article-container">
+                        <div>
+                            <h2 class="title indent">{{ page.title }}</h2>
+                            <p>{{ page.date | date: '%B %d, %Y' }} {{ page.event_time }}</p>
+                            <p>Location: <a href="http://www.uoguelph.ca/campus/map/reynolds/">Reynolds 0002</a></p>
+                            <hr>
+                        </div>
+
+                        <p>{{ content }}</p>
+                        {% if page.files %}
```
Comment (keeferrourke) [inline _layouts/post.html:58]: How does one use the `files` parameter, and what does this end up looking like if its rendered?

File: _layouts/default-legacy.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
```
Comment (keeferrourke) [inline _layouts/default-legacy.html:1]: So this file looks like it's a copy of the `default.html` file in the `_layouts` folder. Is the idea to replace the default layout in a future PR?

File: _layouts/post.html (hunk 1 above)
Comment (BenPearo) [reply _layouts/post.html:58]: This isn't new code, just from the old template. If there's a file added like a PDF it renders a downloads card

File: _layouts/default-legacy.html (hunk 2 above)
Comment (BenPearo) [reply _layouts/default-legacy.html:1]: That's right. The new default has markdown support added, this legacy is there to keep support for the old HTML pages

Comment (keeferrourke) [reply _layouts/default-legacy.html:1]: There isn't a new `default` template though? Am I missing something?

**Review by keeferrourke — APPROVED** [review]

**Pull request merged**